histomicstk/preprocessing/color_deconvolution/color_deconvolution.py

```diff
@@ -68,6 +68,11 @@ def color_deconvolution(im_rgb, w, I_0=None):
 
     """
     # complement stain matrix if needed
+    if w.shape[1] < 3:
```

Comment:
In looking at this, the docs say `w` is array like, so we should probably also add `w = np.array(w)` before we check the shape.